src/main/kotlin/ar/edu/unahur/obj2/servidorWeb/ServidorWeb.kt

```diff
@@ -10,5 +12,58 @@ enum class CodigoHttp(val codigo: Int) {
   NOT_FOUND(404),
 }
 
-class Pedido(val ip: String, val url: String, val fechaHora: LocalDateTime)
+class Pedido(val ip: String, val url: String, val fechaHora: LocalDateTime){
+  fun protocolo() = url.substringBefore(":")
+  fun extension() = url.substringAfterLast(".")
+  fun ruta(): String {
+    val urlSeparated = url.split("/").toMutableList()
+    var i = 0
+    while (i != 3){ urlSeparated.removeFirst();i++}
+    return "/"+urlSeparated.joinToString("/")
+  }
+}
+
+
 class Respuesta(val codigo: CodigoHttp, val body: String, val tiempo: Int, val pedido: Pedido)
+
+class ServidorWeb(val body: String = "",val tiempo: Int = 10){
+  val modulosEstablecidos = mutableListOf<Modulo>()
+  val analizadoresEstablecidos = mutableListOf<Analizador>()
+
+
+  fun agregarModulo(modulo: Modulo) = modulosEstablecidos.add(modulo)
+  fun agregarAnalizador(analizador: Analizador) = analizadoresEstablecidos.add(analizador)
+
+  fun puedeAtender(pedido: Pedido) = pedido.protocolo() == "http"
+
+
+  fun atender(pedido: Pedido): Respuesta {
+    var respuesta = Respuesta(NOT_FOUND,"",10,pedido) // es lo que debe devolver cuando hay error
+    val modulo = this.moduloPara(pedido)!!
+    if(this.hayModuloPara(pedido)){
+      respuesta = Respuesta(OK,modulo.body,modulo.tiempo,pedido)
+      modulo.agregaRespuesta(respuesta)
+    }
+    if(hayAnalizadores()){
+      analizadoresEstablecidos.map { it.agregarModulo(modulo) }
+      analizadoresEstablecidos.map { it.agregarRespuesta(respuesta) }
+    }
```

Comment:
**Acoplamiento**
La idea del patrón Observer es que los objetos del dominio conozcan lo menos posible a los observadores. Acá, por ejemplo, el servidor web sabe que el analizador almacena los módulos y las respuestas. Esto no es necesario para todos los analizadores, por ejemplo, el de demoras solo le interesa contar pero no tener un registro de cada respuesta individual.

Tendría que ser tipo, una sola vez, `analizadores.forEach { it.notificar(modulo, respuesta) }` y que cada analizador haga con eso lo que le  plazca. Si lo quiere guardar en una colección (el caso de estadísticas), buenísimo, pero si solo le interesa sumar a un contador la cantidad de respuestas demoradas (el caso de demoras) hará eso.